frontend/src/metabase/parameters/utils/mapping-options.ts

```diff
@@ -211,3 +213,64 @@ export function getParameterMappingOptions(
 
   return options;
 }
+
+export function getMappingOptionByTarget(
+  mappingOptions: ParameterMappingOption[],
+  target?: ParameterTarget | null,
+  question?: Question,
+  parameter?: Parameter,
+): ParameterMappingOption | undefined {
+  if (!target) {
+    return;
+  }
+
+  const matchedMappingOptions = mappingOptions.filter(mappingOption =>
```

Comment:
Code until `const { query, columns } = getParameterColumns(question, parameter);` is slightly changed to work without dashcards. We now use `isStructuredDimensionTarget` helper which is also used by code that generates ad-hoc queries for dashboards -> QB drill-thru